src/grid/initial_conditions.cpp

```diff
@@ -322,15 +323,15 @@ void Grid3D::Linear_Wave(Real rho, Real vx, Real vy, Real vz, Real P, Real A, Re
         Real x_pos, y_pos, z_pos;
         Get_Position(i_rot, j_rot, k_rot, &x_pos, &y_pos, &z_pos);
 
-        // set constant initial states. Note that mhd::utils::computeEnergy
-        // computes the hydro energy if MHD is turned off
+        // set constant initial states. Note that hydro_utilities::Calc_Energy_Primitive computes the correct MHD or
+        // hydro energy
```

Comment:
I think it's a great idea to have a comment here that points out this subtlety, but I think it would be helpful if it explicitly stated that what determines if `Calc_Energy_Primitive` gives you the hydro or MHD energy is whether or not you're using the hydro or MHD build.

Reply:
Done. See commit 525823e